# OpenBSD has no huge-page mapping flags in this allocator.
# Use anonymous read/write pages and report no large pages.

Index: src/crypto/common/VirtualMemory_unix.cpp
--- src/crypto/common/VirtualMemory_unix.cpp.orig
+++ src/crypto/common/VirtualMemory_unix.cpp
@@ -86,7 +86,7 @@ bool xmrig::VirtualMemory::isHugepagesAv
 {
 #   ifdef XMRIG_OS_LINUX
     return std::ifstream("/proc/sys/vm/nr_hugepages").good() || std::ifstream("/sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages").good();
-#   elif defined(XMRIG_OS_MACOS) && defined(XMRIG_ARM) || defined(XMRIG_OS_HAIKU)
+#   elif defined(XMRIG_OS_MACOS) && defined(XMRIG_ARM) || defined(XMRIG_OS_HAIKU) || defined(XMRIG_OS_OPENBSD)
     return false;
 #   else
     return true;
@@ -158,6 +158,8 @@ void *xmrig::VirtualMemory::allocateExec
     }
 #   elif defined(XMRIG_OS_HAIKU)
     void *mem = mmap(0, size, PROT_READ | PROT_WRITE | PROT_EXEC, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+#   elif defined(XMRIG_OS_OPENBSD)
+    void *mem = mmap(0, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
 #   else
 
     void *mem = nullptr;
@@ -182,7 +184,7 @@ void *xmrig::VirtualMemory::allocateLarg
     void *mem = mmap(0, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, VM_FLAGS_SUPERPAGE_SIZE_2MB, 0);
 #   elif defined(XMRIG_OS_FREEBSD)
     void *mem = mmap(0, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_ALIGNED_SUPER | MAP_PREFAULT_READ, -1, 0);
-#   elif defined(XMRIG_OS_HAIKU)
+#   elif defined(XMRIG_OS_HAIKU) || defined(XMRIG_OS_OPENBSD)
     void *mem = nullptr;
 #   else
     void *mem = mmap(0, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_POPULATE | hugePagesFlag(hugePageSize()), 0, 0);
